subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/DefaultBuildTreeModelControllerServices.kt

```diff
@@ -146,6 +147,21 @@ class DefaultBuildTreeModelControllerServices : BuildTreeModelControllerServices
             registration.add(ProjectScopedScriptResolution::class.java, ProjectScopedScriptResolution.NO_OP)
             registration.addProvider(VintageBuildTreeProvider())
         }
+        if (modelParameters.isIntermediateModelCache) {
+            registration.addProvider(ConfigurationCacheModelProvider())
+        } else {
+            registration.addProvider(VintageModelProvider())
+        }
+    }
+
+    private
+    class ConfigurationCacheModelProvider {
```

Comment:
Same as below. I was just using consistent naming as elsewhere, where this code originated. 

As for using lambdas, I'm. not sure that would work since these methods need to be specially named, starting with `create`, in order to properly work with the service infrastructure. 

There probably is a way around this, but I really don't think its a big deal

Reply:
Not a big deal, especially if naming is required.